### ISSUE TITLE
add shape annotation reordering API

### ISSUE THREAD

**Issue by incanus** [mapbox/mapbox-gl-native#1728]
Punted from #1655. Should be as simple as updating `AnnotationManager::orderedShapeAnnotations`, which affects the order of shape layers returned from `getTile()`.

**Comment by akbsteam** [comment]
is there a timeline for this feature?

**Comment by 1ec5** [comment]
Note that this issue tracks an API for arbitrarily reordering non-view-backed point annotations. We don’t currently have a timeline for this feature, but pull requests are more than welcome.

View-backed annotations automatically come to the front as of mapbox/mapbox-gl-native#5264 (iOS) and mapbox/mapbox-gl-native#5294 (Android). It’s also possible to arbitrarily reorder view-backed annotations on iOS: see https://github.com/mapbox/mapbox-gl-native/pull/5287#issue-159247473 for an example. mapbox/mapbox-gl-native#5238 tracks an API for arbitrarily reordering view-backed annotations on Android.

Meanwhile, in the coming Android SDK v4.2.0 (currently in beta) and iOS SDK v3.4.0 (in alpha), the runtime styling API offers an alternative to annotations. You can arbitrarily order a style layer above or below any other style layer, even beneath parts of the base map like labels.

Hopefully the improvements in view-backed annotations and runtime styling meet your needs. If not, we’d love to hear about your specific use cases, to inform the improvements we make going forward.

**Comment by incanus** [comment]
No, this issue was originally for polygon annotation reordering on the fly.

**Comment by Xgamefactory** [comment]
these are simple and base features that every map sdks have like google,apple mapkit,heremaps event mapbox ios sdk.
